src/Honeycomb.OpenTelemetry/HoneycombExporter.cs

```diff
@@ -20,75 +19,75 @@ public HoneycombExporter(IHoneycombService honeycombService, IOptions<HoneycombA
             _settings = settings;
         }
 
-        public async override Task<ExportResult> ExportAsync(IEnumerable<SpanData> batch, CancellationToken cancellationToken)
+        public override ExportResult Export(in Batch<Activity> batch)
         {
-            var honeycombEvents = new List<HoneycombEvent>();
-
-            foreach (var span in batch)
+            using (var scope = SuppressInstrumentationScope.Begin())
             {
-                if (span.Attributes
-                        .Any(a => a.Key == "http.url" && 
-                                 (a.Value is string urlStr && 
-                                  urlStr.StartsWith("https://api.honeycomb.io/"))))
-                    continue;
-                
-                var events = GenerateEvent(span);
-                honeycombEvents.AddRange(events);
-            }
-            await _honeycombService.SendBatchAsync(honeycombEvents);
+                try
+                {
+                    var honeycombEvents = new List<HoneycombEvent>();
+                    foreach (var activity in batch)
+                    {
+                        // TODO: skip activities where url starts with https://api.honeycomb.com
+
+                        var events = GenerateEvent(activity);
+                        honeycombEvents.AddRange(events);
+                    }
 
-            return ExportResult.Success;
+                    _honeycombService.SendBatchAsync(honeycombEvents).ConfigureAwait(false).GetAwaiter().GetResult();
+                    return ExportResult.Success;
+                }
+                catch (Exception)
+                {
+                    return ExportResult.Failure;
+                }
+            }
         }
 
-        private IEnumerable<HoneycombEvent> GenerateEvent(SpanData span)
+        private IEnumerable<HoneycombEvent> GenerateEvent(Activity activity)
         {
             var list = new List<HoneycombEvent>();
 
             var ev = new HoneycombEvent {
-                EventTime = span.StartTimestamp.UtcDateTime,
+                EventTime = activity.StartTimeUtc,
                 DataSetName = _settings.Value.DefaultDataSet
             };
             var baseAttributes = new Dictionary<string, object> {
-                {"trace.trace_id", span.Context.TraceId.ToString()},
-                {"service_name", span.Name}
+                {"trace.trace_id", activity.Context.TraceId.ToString()},
+                {"service_name", activity.DisplayName}
```

Comment:
Is this correct? why should the activity display name become the `service_name` and what even is `service_name`?

Reply:
https://docs.honeycomb.io/working-with-your-data/home/#fields

Theres the preconfigured fields, it has a few variants.

Reply:
Thanks @martinjt - I read that and I am still a little confused.

I would have thought something like this would be appropriate for the preconfigured field `service_name`?

![image](https://user-images.githubusercontent.com/570470/96521964-cb0dec80-12b5-11eb-9a33-49940898383c.png)

Reply:
Yeah, service name would typically be a consistent process or application name. The activity name would be equivalent to span name.